This PR covers issue #1496 (#1516)
OWASP/owasp-mstg: Document/0x06d-Testing-Data-Storage.md

000200a644d7d2c56eec5b89c1921dacbec83c3e
```

Therefore, it's not straightforward to navigate through it and you will not find any hints of the app you want to analyze in the directory or file name. You can consider using the [iMazing](https://imazing.com "iMazing") shareware utility to assist here. Perform a device backup with iMazing and use its built-in backup explorer to easily analyze app container contents including original paths and file names. Without iMazing or similar software you may need to resort to using grep to identify sensitive data. This is not the most thorough approach but you can try searching for sensitive data that you have keyed in while using the app before you made the backup. For example: the username, password, credit card data, PII or any data that is considered sensitive in the context of the app. 

```bash
$ ~/Library/Application Support/MobileSync/Backup/<UDID>
$ grep -iRn "password" .
```

As described in the Static Analysis section, any sensitive data that you're able to find should be excluded from the backup, encrypted properly by using the Keychain or not stored on the device in the first place. 

> In case you need to work with an encrypted backup, the [following Python scripts (backup_tool.py and backup_passwd.py)](https://github.com/dinosec/iphone-dataprotection/tree/master/python_scripts "iphone-dataprotection") will be a good starting point. They might not work with the latest iTunes versions and might need to be tweaked.

Reviewer: ```suggestion
In case you need to work with an encrypted backup, there are some Python scripts in [DinoSec's GitHub repo](https://github.com/dinosec/iphone-dataprotection/tree/master/python_scripts "iphone-dataprotection"), such as backup_tool.py and backup_passwd.py, that will serve as a good starting point. However, note that they might not work with the latest iTunes versions and might need to be tweaked.
```

As discussed earlier, sensitive data is not limited to just user data and PII. It can also be configuration or settings files that affect app behavior, restrict functionality, or enable security controls. If you take a look at the open source bitcoin wallet app, Bither, you'll see that it's possible to configure a PIN to lock the UI. And after a few easy steps, you will see how to bypass this UI lock with a modified backup on a non-jailbroken device.

<table bordercolor="#FFFFFF">
  <tr><td>
    <img src="Images/Chapters/0x06d/bither_demo_enable_pin.PNG" alt="configure pin" width="270">
  </td><td>
    <img src="Images/Chapters/0x06d/bither_demo_pin_screen.PNG" alt="pin enabled" width="270">
  </td></tr>
</table>

After you enable the pin, use iMazing to perform a device backup:
1. Select your device from the list under the Available menu
2. Click the top menu option ‘Back Up’
3. Follow prompts to complete the backup using defaults

Next you can open the backup to view app container files within your target app
1. Select your device and click Backups on the top right menu
2. Click the backup you created and select 'View'

> 3. Navigate to the Bither app from the Apps directory

Reviewer: ```suggestion
3. Navigate to the Bither app from the **Apps** directory.
```

At this point you can view all the backed up content for Bither. 

<img src="Images/Chapters/0x06d/bither_demo_imazing_1.png" alt="iMazing" width="550">

> This is where you can begin parsing through the files looking for sensitive data. In the screenshot you'll see the net.bither.plist file which contains the pin_code attribute. To remove the UI lock restriction, simply delete the pin_code attribute and save the changes. 

Reviewer: ```suggestion
This is where you can begin parsing through the files looking for sensitive data. In the screenshot you'll see the net.bither.plist file which contains the `pin_code` attribute. To remove the UI lock restriction, simply delete the `pin_code` attribute and save the changes. 
```

> From there it's possible to easily restore the modified version of net.bither.plist back onto the device using the licensed version of iMazing. The free workaround, however, is to find the plist file in the obfuscated backup generated by iTunes. So create your iTunes backup of the device with Bither's pin code configured. Then, using the steps described earlier, find the backup directoy and grep for "pin_code" as shown below.

Reviewer: ```suggestion
From there it's possible to easily restore the modified version of net.bither.plist back onto the device using the licensed version of iMazing. The free workaround, however, is to find the plist file in the obfuscated backup generated by iTunes. So create your iTunes backup of the device with Bither's PIN code configured. Then, using the steps described earlier, find the backup directory and grep for "pin_code" as shown below.
```

```bash
$ ~/Library/Application Support/MobileSync/Backup/<UDID>
$ grep -iRn "pin_code" .
Binary file ./13/135416dd5f251f9251e0f07206277586b7eac6f6 matches
```

You'll see there was a match on a binary file with an obfuscated name. This is your net.bither.plist file. Go ahead and rename the file giving it a plist extension so Xcode can easily open it up for you. 

<img src="Images/Chapters/0x06d/bither_demo_plist.png" alt="iMazing" width="550">

> Again, remove the pin_code attribute from the plist and save your changes. Rename the file back to the original name (i.e., without the plist extenstion) and perform your backup restore from iTunes. When the restore is complete you'll see that Bither no longer prompts you for the pin code when launched.

Reviewer: ```suggestion
Again, remove the `pin_code` attribute from the plist and save your changes. Rename the file back to the original name (i.e., without the plist extension) and perform your backup restore from iTunes. When the restore is complete you'll see that Bither no longer prompts you for the PIN code when launched.
```
